Userland/Libraries/LibC/stdio.cpp

```diff
@@ -261,30 +263,36 @@ bool FILE::gets(u8* data, size_t size)
                 return total_read > 0;
             }
             size_t actual_size = min(size - 1, queued_size);
-            u8* newline = reinterpret_cast<u8*>(memchr(queued_data, '\n', actual_size));
-            if (newline)
-                actual_size = newline - queued_data + 1;
-            memcpy(data, queued_data, actual_size);
-            m_buffer.did_dequeue(actual_size);
+            T const* newline = nullptr;
+            for (size_t i = 0; i < actual_size; ++i) {
+                if (queued_data[i] != '\n')
```

Comment:
Shouldn't this actually be a `L'\n'` if `T` if `wchar_t`?

Reply:
Same value, no difference here.
unless we want to care about having long char literals in any encoding except utf-32.